File: src/11. Container With Most Water/11.py
from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxarea, l, r = 0, 0, len(height) - 1
        while l < r:
            if height[l] < height[r]:
                maxarea = max(maxarea, height[l]*(r-l))
                l += 1
            else:
                maxarea = max(maxarea, height[r]*(r-l))
                r -= 1
        return maxarea


if __name__ == "__main__":
    h = [1, 8, 6, 2, 5, 4, 8, 3, 7]
    print(Solution().maxArea(h))

File: src/11. Container With Most Water/11.java
class Solution {
    public int maxArea(int[] height) {
        int maxarea = 0, l = 0, r = height.length - 1;
        while (l < r) {
            if (height[l] < height[r]) {
                maxarea = Math.max(maxarea, height[l]*(r-l));
                l++;
            } else {
                maxarea = Math.max(maxarea, height[r]*(r-l));
                r--;
            }
        }
        return maxarea;
    }
    public static void main(String[] args) {
        int[] height = {1, 8, 6, 2, 5, 4, 8, 3, 7};
        Solution s = new Solution();
        int ans = s.maxArea(height);
        System.out.println(ans);
        System.out.println(new Solution().maxArea(height));
        
    }
}

